core/storage/blog_statistics/gae_models_test.py

```diff
@@ -0,0 +1,817 @@
+            self.BLOG_POST_TWO_ID)
+
+        self.assertEqual(stats_model.id, self.BLOG_POST_TWO_ID)
+
+        # Test should raise exception if model with given blog post id already
+        # exists.
+        with self.assertRaisesRegex(
+            Exception,
+            'A blog post views stats model with the given blog post ID'
+            'exists already.'):
+            blog_stats_models.BlogPostViewsAggregatedStatsModel.create(
+                 self.BLOG_POST_TWO_ID
+            )
+
+    def test_get_model_association_to_user(self) -> None:
+        self.assertEqual(
+            blog_stats_models.BlogPostViewsAggregatedStatsModel
+                .get_model_association_to_user(),
+            base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+        )
+
+    def test_get_export_policy(self) -> None:
+        expected_export_policy_dict = {
+            'views_by_hour': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'views_by_date': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'views_by_month': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'created_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'last_updated': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'deleted': base_models.EXPORT_POLICY.NOT_APPLICABLE
+        }
+        self.assertEqual(
+            blog_stats_models.BlogPostViewsAggregatedStatsModel
+                .get_export_policy(),
+            expected_export_policy_dict
+        )
+
+
+class BlogPostReadsAggregatedStatsModelUnitTests(test_utils.GenericTestBase):
+    """Test the BlogPostReadsAggregatedStatsModel class."""
+
+    BLOG_POST_ONE_ID: Final = 'blog_post_one'
+    BLOG_POST_TWO_ID: Final = 'blog_post_two'
+
+    def setUp(self) -> None:
+        """Set up blog post reads aggregated stats models in datastore for use
+        in testing.
+        """
+        super().setUp()
+
+        self.model_one = (
+            blog_stats_models.BlogPostReadsAggregatedStatsModel(
+                id=self.BLOG_POST_ONE_ID,
+                reads_by_hour={},
+                reads_by_date={},
+                reads_by_month={}
+            )
+        )
+        self.model_one.update_timestamps()
+        self.model_one.put()
+
+    def test_get_deletion_policy(self) -> None:
+        self.assertEqual(
+            blog_stats_models.BlogPostReadsAggregatedStatsModel
+            .get_deletion_policy(),
+            base_models.DELETION_POLICY.NOT_APPLICABLE
+        )
+
+    def test_create_and_get_stats_models(self) -> None:
+        blog_stats_models.BlogPostReadsAggregatedStatsModel.create(
+            self.BLOG_POST_TWO_ID
+        )
+        stats_model = blog_stats_models.BlogPostReadsAggregatedStatsModel.get(
+            self.BLOG_POST_TWO_ID)
+
+        self.assertEqual(stats_model.id, self.BLOG_POST_TWO_ID)
+
+        # Test should raise exception if model with given blog post id already
```

Comment:
Comments should be used sparingly because ideally your code itself should be able to explain what's happening. For e.g. by breaking down the functions into smaller pieces or using more descriptive function/variable names. Lots of comments create an overhead for who ever will makes changes in the code next. In this case, the exception message makes it clear an exception is being raised because the blog post id already exists. 

Same change for all other comments. See if your code can make it clear to the reader what's happening. Only rely on a comment if that is not true.

Reply:
Done